tools/mksurfdata_esmf/src/mksoiltexMod.F90

```diff
@@ -98,237 +115,428 @@ subroutine mksoiltex(file_mesh_i, file_data_i, mesh_o, pioid_o, pctlnd_pft_o, rc
-       write (ndiag,'(a)') 'The actual data is continuous %sand, %silt and %clay not textural classes'
-    end if
-    call output_diagnostics_index(mesh_i, mesh_o, mask_i, frac_o, &
-         0, nlsm, soil_i, soil_o, 'soil texture class',  ndiag, rc)
-    if (chkerr(rc,__LINE__,u_FILE_u)) return
+          ! Same algorithm for bulk_o as for sand_o
+          bulk_o(no,1) = bulk_i(1,1,lookup_index)
+          if (bulk_o(no,1) < 0._r4) then
+             do l = 2,n_scid
+                if (bulk_i(1,l,lookup_index) >= 0._r4) then
+                   bulk_o(no,1) = bulk_i(1,l,lookup_index)
+                   exit
+                end if
+             end do
+          end if
+          if (bulk_o(no,1) < 0._r4) then
+             if (int(bulk_o(no,1)) == -4) then  ! sand dunes
+                write(6,'(a,i8)')'WARNING: changing bulk_o from -4 to 1 at no = ',no
+                bulk_o(no,:) = 1.5_r4  ! TODO Ok for sand dunes?
+             else
+                write(6,'(a,i8,a,i8)')'WARNING: changing bulk_o from ',int(bulk_o(no,1)),' to 0 at no = ',no
+                bulk_o(no,:) = 1.5_r4  ! TODO Ok for -7?
+             end if
+          end if
+          if (bulk_o(no,1) < 0._r4) then
+             write(6,*)'ERROR: at no, lookup_index = ',no,lookup_index
+             call shr_sys_abort('could not find a value >= 0 for bulk_i')
+          end if
+          do l = 2,nlay
+             bulk_o(no,l) = bulk_i(l,1,lookup_index)
+             if (bulk_o(no,l) < 0._r4) then
+                bulk_o(no,l) = bulk_o(no,l-1)
+             end if
+          end do
+
+          ! Same algorithm for phaq_o as for sand_o
+          phaq_o(no,1) = phaq_i(1,1,lookup_index)
+          if (phaq_o(no,1) < 0._r4) then
+             do l = 2,n_scid
+                if (phaq_i(1,l,lookup_index) >= 0._r4) then
+                   phaq_o(no,1) = phaq_i(1,l,lookup_index)
+                   exit
+                end if
+             end do
+          end if
+          if (phaq_o(no,1) < 0._r4) then
+             if (int(phaq_o(no,1)) == -4) then  ! sand dunes
+                write(6,'(a,i8)')'WARNING: changing phaq_o from -4 to 1 at no = ',no
+                phaq_o(no,:) = 7._r4
+             else
+                write(6,'(a,i8,a,i8)')'WARNING: changing phaq_o from ',int(phaq_o(no,1)),' to 0 at no = ',no
+                phaq_o(no,:) = 7._r4
+             end if
+          end if
+          if (phaq_o(no,1) < 0._r4) then
+             write(6,*)'ERROR: at no, lookup_index = ',no,lookup_index
+             call shr_sys_abort('could not find a value >= 0 for phaq_i')
+          end if
+          do l = 2,nlay
+             phaq_o(no,l) = phaq_i(l,1,lookup_index)
+             if (phaq_o(no,l) < 0._r4) then
+                phaq_o(no,l) = phaq_o(no,l-1)
+             end if
+          end do
+
+          ! TODO Rm organic before merging PR #1732 UNLESS we opt for OPTION 2
+          ! organic_o OPTION 1: as we plan to calculate organic in the CTSM
```

Comment:
As a test I calculated organic_o in two different ways...
- Option 1 as we discussed this morning: calculate organic in the CTSM from orgc_o, cfrag_o, and bulk_o (i.e. after the terms have been regridded).
- Option 2 (a few lines up where orgc_o is also calculated): In this case I regrid organic_i (calculated from orgc_i, cfrag_i, and bulk_i) to organic_o. This may be the more accurate approach. It goes against this morning's idea to keep ORGC the same in fsurdat as in the raw data, but it may be more accurate because it first calculates organic_i and then regrids to organic_o rather than regridding all the terms first and then calculating organic_o.

The fsurdat files from this test are in `/glade/work/slevis/git/mksurfdata_toolchain/tools/mksurfdata_esmf`:
```
surfdata_1.9x2.5_hist_78pfts_CMIP6_1850_c220527_OPT1.nc
surfdata_1.9x2.5_hist_78pfts_CMIP6_1850_c220527_OPT2.nc
surfdata_1.9x2.5_hist_78pfts_CMIP6_1850_c220527_OPT2-1.nc (the difference file)
```
Also, the old fsurdat that I'm comparing against:
`/glade/p/cesmdata/inputdata/lnd/clm2/surfdata_map/release-clm5.0.18/surfdata_1.9x2.5_hist_78pfts_CMIP6_simyr1850_c190304.nc`
I'm happy to follow up next week. Have a nice holiday weekend!

Reply:
Posting the difference in ORGANIC calculated by method 2 vs. 1.
Very little difference, I think because we regrid by most dominant rather than by averaging.

![organic2vs1](https://user-images.githubusercontent.com/35322900/170874973-bd6cf78b-1a28-4625-b25a-a772406f63e2.JPG)

Reply:
When I get a chance, I'm curious to understand this small difference. Meanwhile though I've confirmed what should go without saying, i.e. that ORGC, CFRAG, BULK, PHAQ (and all other fields) are identical between these two fsurdats.

Reply:
Oh and old fsurdat shows max organic = 130 as we discussed:

![organic_old](https://user-images.githubusercontent.com/35322900/170875584-747729f6-9a1d-4a73-8681-5be47df410ea.JPG)